pkg/app/flag.go

```diff
@@ -0,0 +1,51 @@
+package app
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	flagsDir = "/tmp/flags"
+	// LimitedCache represent the flag cache couldn't mount on ssd or hdd
+	LimitedCache = "limited-cache"
+)
+
+// SetFlag is used when the /var/cache cannot be mounted on a SSD or HDD,
+// it will mount the cache disk on a temporary file system in memory.
+func SetFlag(key string) error {
+	// creating a file that will be used as a flag
+	// the flag can 'warn' other deamons that the /var/cache is not on HDD or SDD
+	if err := os.MkdirAll(flagsDir, os.ModePerm); err != nil {
+		return errors.Wrap(err, "failed to create flags directory")
+	}
+
+	f, err := os.Create(filepath.Join(flagsDir, key))
+	if err != nil {
+		return errors.Wrap(err, "failed to create the flag file")
+	}
+
+	return f.Close()
+}
+
+// CheckFlag checks the status of a flag based on a key
+func CheckFlag(key string) bool {
+	_, err := os.Stat(filepath.Join(flagsDir, key))
+	return !os.IsNotExist(err)
+}
+
+// DeleteFlag deletes (unsets) a given flag based on a key
+func DeleteFlag(key string) error {
+	d, err := os.Open(flagsDir)
```

Comment:
Why do u open the directory here ?

Reply:
added check to make sure directory exist, in case the code would execute when there is no flag set. A "just-to-be-sure". Should it be removed?

Reply:
First of all, a check if the directory exists would be more efficient if u do `os.Stat` on the file, not a Open, then close. Secondly, a RemoveAll call does not fail if the file does not exist in the first place, so a check is not really needed at all.

Reply:
fixed